docs/tutorials/geomesa-authorizations.rst

```diff
@@ -0,0 +1,812 @@
+
+.. code-block:: java
+
+    com.example.geomesa.authorizations.LdapAuthorizationsProvider
+    com.example.geomesa.authorizations.LdapAuthorizationsProviderTest
+
+There is a class that shows how to query GeoServer through WFS that will
+be explored in more detail later in the tutorial:
+
+.. code-block:: java
+
+    com.example.geomesa.authorizations.GeoServerAuthorizationsTutorial
+
+Additionally, there are two helper classes included in the tutorial:
+
+-  ``com.example.geomesa.authorizations.GdeltFeature`` - Contains the attributes available
+   in the GDELT data set.
+-  ``com.example.geomesa.authorizations.SetupUtil`` - Handles reading command-line
+   arguments
+
+Applying Authorizations and Visibilities to GeoServer Using PKIs And LDAP
+-------------------------------------------------------------------------
+
+This section will show you how to configure GeoServer to authenticate
+users with PKIs, use LDAP to store authorizations, then apply
+authorizations on a per-user/per-query basis.
+
+Basic user authentication will take place via user certificates. Each
+user will have their own public/private key pair that uniquely
+identifies them.
+
+User authorizations will come from LDAP. Once a user's identity has been
+verified via PKI, we will look up the user's details in LDAP.
+
+Once we have a user's authentication and authorizations, we will apply
+them to the GeoMesa query using a custom ``AuthorizationsProvider``
+implementation.
+
+.. note:: 
+
+    It is assumed for the rest of the tutorial that you have created
+    the GeoServer data stores and layers outlined in the GDELT
+    tutorial </geomesa-gdelt-analysis/>
+
+Run GeoServer in Tomcat
+~~~~~~~~~~~~~~~~~~~~~~~
+
+*Note: If you are already running GeoServer in Tomcat, you can skip this
+step.*
+
+GeoServer ships by default with an embedded Jetty servlet. In order to
+use PKI login, we need to install it in Tomcat instead.
+
+1. Download and install Tomcat 7.
+2. Create an environment variable pointing to your Tomcat installation (you
+   may want to add this to your bash init scripts):
+
+.. code-block:: bash
+
+    export CATALINA_HOME=/path/to/tomcat
+
+3. If you want to reuse your existing GeoServer configuration, create an
+   environment variable pointing to your GeoServer data directory (you may
+   want to add this to your shell initialization scripts):
+
+.. code-block:: bash
+
+    export GEOSERVER_DATA_DIR=/path/to/geoserver/data_dir
+
+4. Copy the GeoServer webapp from the GeoServer distribution into the
+   tomcat servlet:
+
+.. code-block:: bash
+
+    cp -r /path/to/geoserver/webapps/geoserver/ $CATALINA_HOME/webapps/
+
+5. Increase the memory allocated to Tomcat, which you will need for running
+   complex queries in GeoServer (the values here may not be applicable for
+   every installation):
+
+.. code-block:: bash
+
+    cd $CATALINA_HOME/bin
+    echo 'CATALINA_OPTS="-Xmx2g -XX:MaxPermSize=128m"' >> setenv.sh
```

Comment:
probably set MaxPermSize to 512M and setting XmX to 6g bc you will run out of memory with 2

also add headless param